### PR TITLE
added support for multiple graphs

### PR DESCRIPTION
**Pull request by nishaq503**
Manifold now contains a graphs variable.
There is one graph per Selection Criterion provided for Manifild.build

### CODE REVIEW

File: pyclam/criterion.py [hunk 1]
```diff
@@ -263,146 +256,3 @@ def _select(root: Cluster, upper: float, lower: float, grace_depth: int) -> Set[
...
-            and graph.subsumed_edges[cluster])
-    }
-    remainder: Set[Cluster] = {cluster for cluster in graph.clusters if cluster not in removals}
-    additions: Set[Cluster] = set()
-    [additions.update(cluster.children) for cluster in removals]
-
-    return remainder.union(additions)
-
-
-class MinimizeSubsumed(GraphCriterion):
```
Comment (thoward27) [inline pyclam/criterion.py:319]: Are we sure we want to get rid of these criteria?

File: pyclam/manifold.py [hunk 2]
```diff
@@ -262,11 +261,15 @@ def descendents(self) -> List['Cluster']:
...
         if 'ratios' not in self.cache:
-            self.cache['ratios'] = [  # Child/Parent Ratios
-                self.local_fractal_dimension / self.parent.local_fractal_dimension,  # local fractal dimension
-                self.cardinality / self.parent.cardinality,  # cardinality
-                max(self.radius, 1e-16) / max(self.parent.radius, 1e-16)  # radius
-            ] if self.depth > 0 else [1., 1., 1.]
+            if self.depth > 0:
+                lfd_ratio = self.local_fractal_dimension / self.parent.local_fractal_dimension
+                self.cache['ratios'] = [  # Child/Parent Ratios
+                    1 / (1 + np.exp(-lfd_ratio)),  # local fractal dimension
```
Comment (thoward27) [inline pyclam/manifold.py:267]: interesting to just do this for one ratio

File: pyclam/manifold.py [hunk 3]
```diff
@@ -1198,7 +1205,8 @@ def distance(self, x1: Union[List[int], Data], x2: Union[List[int], Data]) -> np
 
     def build(self, *criteria) -> 'Manifold':
         """ Rebuilds the Cluster-tree and the Graph-stack. """
-        from pyclam.criterion import ClusterCriterion, SelectionCriterion, GraphCriterion
+        from pyclam.criterion import ClusterCriterion, SelectionCriterion
```
Comment (thoward27) [inline pyclam/manifold.py:1208]: weird to lose graph criterion in build

File: pyclam/manifold.py [hunk 4]
```diff
@@ -1342,7 +1333,7 @@ def dump(self, fp: Union[BinaryIO, IO[bytes]]) -> None:
         pickle.dump({
             'metric': self.metric,
             'root': self.root.json(),
-            'graph': [cluster.name for cluster in self.graph.clusters]
+            'graphs': [[cluster.name for cluster in graph.clusters] for graph in self.graphs]
```
Comment (thoward27) [inline pyclam/manifold.py:1336]: Consider a set for the inner comprehension

File: pyclam/criterion.py (hunk 1 above)
Comment (nishaq503) [reply pyclam/criterion.py:319]: They are alive on a different branch where I'm trying to fix add and remove for Graph

File: pyclam/manifold.py (hunk 2 above)
Comment (nishaq503) [reply pyclam/manifold.py:267]: doing this significantly improved the quality of fit of the linear regression models. The other ratios don't need this

File: pyclam/manifold.py (hunk 3 above)
Comment (nishaq503) [reply pyclam/manifold.py:1208]: Yeah the names probably need some improvement. GraphCriterion are for fine-tuning of graphs. SelectionCriterion are for building the actual graphs. GraphCriterion will make a comeback one I fix add and remove for Graphs

File: pyclam/manifold.py (hunk 4 above)
Comment (nishaq503) [reply pyclam/manifold.py]: done

**Pull request merged**